### ISSUE TITLE
False-positive unreachable code for error-emitting generator

### ISSUE THREAD

**Issue by skeggse** [pylint-dev/pylint#8909]
### Bug description

Per #4698:

> The most efficient way to create a empty generator is to add yield after return. pylint should not add the warning unreachable code for it. https://stackoverflow.com/a/61496399

```py
def func():
    raise Exception("failure")
    yield
```

### Command used

```shell
pylint test.py
```

### Pylint output

```shell
************* Module test
test.py:1:0: C0114: Missing module docstring (missing-module-docstring)
test.py:1:0: C0116: Missing function or method docstring (missing-function-docstring)
test.py:3:4: W0101: Unreachable code (unreachable)
test.py:2:4: W0719: Raising too general exception: Exception (broad-exception-raised)
```

### Expected behavior

```shell
************* Module test
test.py:1:0: C0114: Missing module docstring (missing-module-docstring)
test.py:1:0: C0116: Missing function or method docstring (missing-function-docstring)
test.py:2:4: W0719: Raising too general exception: Exception (broad-exception-raised)
```

### Pylint version

```shell
pylint 2.17.5
astroid 2.15.6
Python 3.11.2 (main, Feb 16 2023, 03:15:23) [Clang 14.0.0 (clang-1400.0.29.202)]
```

**Comment by Pierre-Sassoulas** [comment]
You're not returning though, you're raising an exception, do:
```python
def func():
    return
    yield
```
The correct snippet does not raise since pylint 2.9.4.

**Comment by skeggse** [comment]
@Pierre-Sassoulas this request is about raising an exception, not about returning early. These produce different behavior, and I'm only referencing #4698 because the language is pretty relevant.

Observe:

```py
def func_gen():
    raise Exception("failure")
    yield


def func_nogen():
    raise Exception("failure")

func_gen() # produces <generator object func_gen at 0x10c4fc820>

func_nogen() # raises Exception: a
```


Including the `yield` makes a material difference in the functionality implemented, and therefore makes the reporting of the `yield` as `unreachable` a false positive.

Please re-open.
